feat(KNO-13285): final preference center docs (#1528)
knocklabs/docs: content/preferences/hosted-preference-center.mdx

section: Preferences
---

Knock's hosted preference center is a page where your users manage their `default` notification preferences. Knock hosts and renders the page, so you can launch a preference center without building or hosting any UI yourself.

- **Knock hosts the page.** Users access their preferences through a signed URL.

> - **Configure in the dashboard.** Set rows, labels, and branding under **Platform** > **Preferences** > **Preference center**.

Reviewer: On smaller screens, the bolding here is a little overwhelming (the line wraps). In a follow up, maybe this would be better as a table?

> - **Configure in the dashboard.** Set rows, labels, and branding under **Platform** > **Preferences** > **Preference center**.

Reviewer 2: ```suggestion
- **Configure in the dashboard.** Set rows, labels, and branding under <a href="https://dashboard.knock.app/~/preferences?tab=preference_center" target="_blank">**Platform** > **Preferences** > **Preference center**</a>.
```
We should add a direct link to the dashboard here

> - **No code required.** Enable the preference center and link to it from your notifications.

Reviewer: In a follow up, this might  benefit from a link on "link to it from your notifications"

> - **User default preferences only.** Does not support [per-tenant](/multi-tenancy/per-tenant-preferences), [object](/preferences/object-preferences), or non-default preference sets.

Reviewer: in follow up,  think we should change the wording here "Does not.." to "The hosted preference center does not.."

> This is the fastest way to let your users update their preferences when they receive a notification. If you need more control over the experience, see the [custom preference center](/preferences/custom-preference-center). Not sure which approach fits? See [Going live with preferences](/preferences/overview#going-live-with-preferences).

Reviewer: (nit) In follow up, I think we should rewrite the first line here. IMO better positioning is along the lines of: The hosted preference center enables teams to quickly give users opt out options.." (e.g. it's not just about speed - a lot of customers who have hit this friction previously simply didn't have eng resources to pursue an alternative. Many customers will be better serves by a full preference center, so I don't think we should purely focus on speed to launch here.)

## Linking to the preference center

> ### From workflow templates

Reviewer 2: ```suggestion
### From a notification
```

Knock renders the user-specific link in the notification. Use `vars.manage_preferences_url` when you want to give users full control over their preferences via the hosted preference center. For single-click opt-out of commercial messages via the commercial unsubscribe feature, use `vars.commercial_unsubscribe_url` instead.

> ### Get a signed URL from your backend

Reviewer: (can be fast follow) I think this section needs some adjustment: The two subheaders are misaligned and a little confusing. Maybe they should be: In workflow templates and In your application?

> ### Get a signed URL from your backend

Reviewer 2: ```suggestion
### From a generated link
```

> Use this when you need to generate a preference center link outside of workflow templates—for example, from your app settings page or an admin panel.

Reviewer 2: ```suggestion
If you need to link a user to their preference center from outside of a notification (for example, from your app settings page or an admin panel), you can generate a link to the hosted preference center.
```

Call [`POST /v1/users/{user_id}/preference_center/signed_url`](/api-reference/users/preference_center/generate_signed_url) from your backend with a **secret API key**. The response includes:

- **`url`.** The full hosted preference center URL for the user.

```bash title="Generate a preference center signed URL"
curl -X POST "https://api.knock.app/v1/users/user_id/preference_center/signed_url" \
  -H "Authorization: Bearer $KNOCK_API_KEY" \
  -H "Content-Type: application/json"
```

> Redirect the user to `url`. If your environment uses a [custom preference center domain](/preferences/hosted-preference-center#custom-domain-for-the-preference-center), the returned `url` reflects that domain.

Reviewer 2: ```suggestion
If your environment uses a [custom preference center domain](/preferences/hosted-preference-center#custom-domain-for-the-preference-center), the returned `url` reflects that domain.
```
Cutting just because this seems superfluous
